core/modules/widgets/select.js

```diff
@@ -51,7 +51,12 @@ SelectWidget.prototype.render = function(parent,nextSibling) {
 Handle a change event
 */
 SelectWidget.prototype.handleChangeEvent = function(event) {
-	var value = this.getSelectDomNode().value;
+	if(this.selectMultiple == false) {
+		var value = this.getSelectDomNode().value;
+	} else {
+		var value = this.getSelectValues()
```

Comment:
noooo. way to complicated and error prone to read and maintain!

I'd prefer something like this:

```
SelectWidget.prototype.handleChangeEvent = function(event) {
    var value;
    if () {
        value = ...
    }
```